java/core/src/java/org/apache/orc/impl/ZstdCodec.java

```diff
@@ -0,0 +1,252 @@
+        throw new IllegalArgumentException(
+            String.format(
+                "Zstd compression window size should be in the range %d to %d,"
+                    + " or set to the default value of 0.",
+                Zstd.windowLogMin(),
+                Zstd.windowLogMax()));
+      }
+      windowLog = newValue;
+      return this;
+    }
+
+    /**
+     * Sets the Zstandard compression codec compression level directly using
+     * the integer setting. This value is typically between 0 and 22, with
+     * larger numbers indicating more aggressive compression and lower speed.
+     * <p>
+     * This method provides additional granularity beyond the setSpeed method
+     * so that users can select a specific level.
+     *
+     * @param newValue The level value of compression to set.
+     * @return ZstdOptions
+     */
+    public ZstdOptions setLevel(int newValue) {
+      if (newValue < Zstd.minCompressionLevel() || newValue > Zstd.maxCompressionLevel()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Zstd compression level should be in the range %d to %d",
+                Zstd.minCompressionLevel(),
+                Zstd.maxCompressionLevel()));
+      }
+      level = newValue;
+      return this;
+    }
+
+    @Override
+    public ZstdOptions setData(DataKind newValue) {
+      return this; // We don't support setting DataKind in ZstdCodec.
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+
+      ZstdOptions that = (ZstdOptions) o;
+
+      if (level != that.level) return false;
+      return windowLog == that.windowLog;
+    }
+
+    @Override
+    public int hashCode() {
+      int result = level;
+      result = 31 * result + windowLog;
+      return result;
+    }
+  }
+
+  private static final ZstdOptions DEFAULT_OPTIONS =
+      new ZstdOptions(1, 0);
+
+  @Override
+  public Options getDefaultOptions() {
+    return DEFAULT_OPTIONS;
+  }
+
+  /**
+   * Compresses an input ByteBuffer into an output ByteBuffer using Zstandard
+   * compression. If the maximum bound of the number of output bytes exceeds
+   * the output ByteBuffer size, the remaining bytes are written to the overflow
+   * ByteBuffer.
+   *
+   * @param in       the bytes to compress
+   * @param out      the compressed bytes
+   * @param overflow put any additional bytes here
+   * @param options  the options to control compression
+   * @return ZstdOptions
+   */
+  @Override
+  public boolean compress(ByteBuffer in, ByteBuffer out,
+      ByteBuffer overflow,
+      Options options) throws IOException {
+    ZstdOptions zso = (ZstdOptions) options;
+
+    zstdCompressCtx = new ZstdCompressCtx();
+    zstdCompressCtx.setLevel(zso.level);
+    zstdCompressCtx.setLong(zso.windowLog);
+    zstdCompressCtx.setChecksum(false);
```

Comment:
For the record, the default value is `false` already.
- https://javadoc.io/static/com.github.luben/zstd-jni/1.5.5-11/com/github/luben/zstd/ZstdCompressCtx.html#setChecksum(boolean)